src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -442,25 +442,39 @@ public async Task ResyncAsync()
 
             if ((peer != null) && (peer.State == NetworkPeerState.HandShaked))
             {
-                var headersPayload = new GetHeadersPayload()
+                var headersPayload = this.BuildGetHeadersPayload();
```

Comment:
I wonder if this is correct though, this is about sending headers, we should be ok to send normal headers to none PH nodes (legacy X)

Reply:
The logic is if the peer is PH we send PH headers is enabled else we send regular headers (regardless of peer being whitelisted) that means X can still sync form #

Reply:
But here you are requesting headers, not sending, am I wrong?

Reply:
So my logic was
If we are connected to a legacy peer, we don't have to request normal headers to that peer except if he's whitelisted.

On the other hand If a legacy peer ask us blocks, we can serve it and it should work because we rely on 
```
// Rely on base.OnMessageReceivedAsync only if the message hasn't be already processed.
await base.OnMessageReceivedAsync(peer, message).ConfigureAwait(false);
```

that actually on my implementation can just handle the case
```
case GetHeadersPayload getHeaders:
   await this.ProcessGetHeadersAsync(peer, getHeaders).ConfigureAwait(false);
   break;
```

this mean that if a peer asks us legacy headers sending us a `getheaders` we serve it in a legacy way

Reply:
Fixed this problem.